smoke-tests-otel-starter/spring-boot-3.2/build.gradle.kts

```diff
@@ -17,6 +17,7 @@ dependencies {
   implementation("org.apache.commons:commons-dbcp2")
   implementation("org.springframework.kafka:spring-kafka")
   implementation("org.springframework.boot:spring-boot-starter-data-mongodb")
+  implementation("org.springframework.boot:spring-boot-starter-aop")
```

Comment:
does it make sense to include this in our starter somehow(?) so this core feature works without users needing to remember to add this extra dependency?

Reply:
great catch - I created https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11775 to fix this

Reply:
As a user,  I would be surprised if the starter added a Spring Boot dependency.

Reply:
starters should be a one-stop solution, where everything you typically need is included, including other starters

users can exclude dependencies or use the autoconfigure module if the starter doesn't fit

so the question is if the annotation is a typical use of the starter

Reply:
> so the question is if the annotation is a typical use of the starter

I think the answer to this is yes

but I don't know what the solution is (if any)

> As a user, I would be surprised if the starter added a Spring Boot dependency.

@jeanbisutti why is this surprising? thanks

Reply:
@zeitlinger @trask The goal of this PR is to make the OTel annotations work in native mode for the OTel Spring starter. The spring-boot-starter-aop change is not related to this PR. So, I propose to mark this conversation as resolved in this PR and reply in #11775.

Reply:
https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11775#discussion_r1672309913